### PR TITLE
Update devcontainer dockerfile

### PR DESCRIPTION
**Pull request by TimOsahenru**
## Issue Link 🔗:

**Issue**: #606 

## Type of Change

- [ ] Bug fix 🐞
- [ ] New feature/page
- [ ] Documentation update
- [x] Other

## Description 📋

Enhanced docker devcontainer to be a base installation of Python and Ruby and then run pip install and ruby bundle and playwright install with dependencies and also included more descriptive comment messages

## Checklist ✅

- [x] Followed the [Code of Conduct](https://github.com/BlackPythonDevs/blackpythondevs.github.io?tab=coc-ov-file) and [Contribution Guide](https://github.com/BlackPythonDevs/blackpythondevs.github.io/blob/469afd71943738b7209bfe56d5a4e35394fbaafc/CONTRIBUTING.md)
- [x] Ran `pre-commit run --all`
- [x] All tests pass locally
- [ ] Added tests (if applicable)
- [ ] Documentation updated (if applicable)

## Additional Notes & Screenshots

Add any additional notes or comments that might be helpful for the reviewers.

### CODE REVIEW

File: .github/.devcontainer/Dockerfile [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
-LABEL org.opencontainers.image.description "Black Python Developers website."
+LABEL org.opencontainers.image.description "Black Python Developers website with pre-installed dependencies."
 ARG RUBY_VERSION=3.2.2
 
+# Install essential tools, Python, pip, and Node.js
 RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
-    && apt-get install -y build-essential \
+    && apt-get install -y \
+        build-essential \
+        python3.11 \
```
Comment (kjaymiller) [inline .github/.devcontainer/Dockerfile:11]: is there a reason we pinned to 3.11?

**Review by kjaymiller — APPROVED** [review]
Great work on this! I had a minor comment but it's not a blocking issue. Either update or merge. 

Again excellent job @TimOsahenru

**Comment by kjaymiller** [comment]
Small nitpick...

It seems that you stacked your changes. This means that you need to merge only one and all the changes will apply. 

Please only merge the last PR otherwise you may run into merge conflicts. Close the others

File: .github/.devcontainer/Dockerfile (hunk 1 above)
Comment (TimOsahenru) [reply .github/.devcontainer/Dockerfile:11]: No specific reason for choosing Python 3.11— default choice. I'm open to switching to 3.13 or another version if needed

**Pull request merged**